### PR TITLE
MHV-56516 Legacy SM Session policy feature flag cleanup

### PR DESCRIPTION
**Pull request by oleksii-morgun**
## Summary

- *This work is behind a feature toggle (flipper): `NO`
- cleanup of a no longer used feature flag for MHV SM session policy

## Related issue(s)

- https://github.com/department-of-veterans-affairs/vets-api/pull/15889
- https://github.com/department-of-veterans-affairs/vets-api/pull/16130

## Testing done

- [x] *New code is covered by unit tests*
- specs updated

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
MyHealth::V1 Messaging controllers

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by mmoyer-va — APPROVED** [review]
Looks good!

**Review by rjohnson2011 — APPROVED** [review]

File: app/services/users/services.rb [hunk 1]
```diff
@@ -18,7 +18,7 @@ def initialize(user)
     #
     def authorizations
       @list << BackendServices::RX if user.authorize :mhv_prescriptions, :access?
-      @list << BackendServices::MESSAGING if user.authorize messaging_service, :access?
+      @list << BackendServices::MESSAGING if user.authorize :mhv_messaging, :access?
```
Comment (dickdavis) [inline app/services/users/services.rb:21]: ❓ The associated policy looks like it'll make an HTTP request each time to determine if the user has access. Do you have an idea of how long the request takes to complete?

Comment (bosawt) [reply app/services/users/services.rb:21]: @dickdavis This is a valid concern, actually it was something I had brought up earlier (see my comments in: https://github.com/department-of-veterans-affairs/vets-api/pull/15889). However as far as I know those concerns were already accommodated, or at least are being watched. 

This PR is just cleaning up a feature flag that's already enabled as far as I can tell, basically the HTTP request is already happening

**Review by bosawt — APPROVED** [review]
Looks good. I assume the flipper is 100% turned on already in production so my assumption is this is a cleanup, with no functional changes

File: app/services/users/services.rb (hunk 1 above)
Comment (oleksii-morgun) [reply app/services/users/services.rb:21]: @dickdavis my understanding is that `MHVSessionBasedClient` should be handling that and using data from cache if a previous request has been made and is not yet expired

**Comment by oleksii-morgun** [comment]
> Looks good. I assume the flipper is 100% turned on already in production so my assumption is this is a cleanup, with no functional changes

@bosawt yes, we had it introduced gradually and it's been at 100% for the last few weeks with no issues raised

**Review by dickdavis — APPROVED** [review]

**Pull request merged**